js/apps/system/_admin/aardvark/APP/react/src/components/jsonEditor/ControlledJSONEditor.tsx

```diff
@@ -13,30 +13,115 @@ const useResetSchema = ({ schema, ajv }: { schema: any; ajv?: any }) => {
   }, [schema, ajv]);
 };
 
-export const ControlledJSONEditor = ({
-  value,
-  onChange,
-  schema,
-  isDisabled,
-  htmlElementProps,
-  ...rest
-}: JsonEditorProps & {
+type ControlledJSONEditorProps = Omit<JsonEditorProps, "value"> & {
+  value?: any;
   isDisabled?: boolean;
+  mainMenuBar?: boolean;
+  isReadOnly?: boolean;
+  defaultValue?: any;
+};
+
+export const ControlledJSONEditor = React.forwardRef(
+  (
+    {
+      value,
+      onChange,
+      schema,
+      isDisabled,
+      isReadOnly,
+      htmlElementProps,
+      mainMenuBar,
+      defaultValue,
+      ...rest
+    }: ControlledJSONEditorProps,
+    ref: React.ForwardedRef<JsonEditor>
+  ) => {
+    const jsonEditorRef = useRef<JsonEditor | undefined>();
+    useResetSchema({ schema, ajv: rest.ajv });
+    useSetupReadOnly({ jsonEditorRef, isDisabled, isReadOnly });
+    useSetupValueUpdate({ jsonEditorRef, value });
+    useSetupDefaultValueUpdate({ jsonEditorRef, defaultValue });
+    return (
+      <ClassNames>
+        {({ css }) => (
+          <JsonEditor
+            schema={schema}
+            ref={node => {
+              if (node) {
+                jsonEditorRef.current = node;
+              }
+              if (typeof ref === "function") {
+                ref(node);
+              } else if (ref) {
+                ref.current = node;
+              }
+            }}
+            value={value || {}}
+            onChange={onChange}
+            htmlElementProps={{
+              ...htmlElementProps,
+              className: isDisabled
+                ? css`
+                    opacity: 0.6;
+                  `
+                : ""
+            }}
+            {...rest}
+            // @ts-ignore
+            mainMenuBar={mainMenuBar}
```

Comment:
That is not possible, hence we needed hack. The underlying JsonEditor npm package doesn't have typed, but it spreads all props, so that makes it work.